miio/huizuo.py

```diff
@@ -147,7 +311,288 @@ def set_brightness(self, level):
     )
     def set_color_temp(self, color_temp):
         """Set color temp in kelvin."""
-        if color_temp < 3000 or color_temp > 6400:
+
+        # I don't know why only one lamp has smaller color temperature (based on specs),
+        # but let's process it correctly
+        if self.model == MODELS_WITH_FAN_WY2:
+            max_color_temp = 5700
+        else:
+            max_color_temp = 6400
+
+        if color_temp < 3000 or color_temp > max_color_temp:
             raise HuizuoException("Invalid color temperature: %s" % color_temp)
 
         return self.set_property("color_temp", color_temp)
+
+
+class HuizuoLampFan(Huizuo):
+    """Support for Huizuo Lamps with fan"""
+
+    # The next section contains the fan management commands
+    # Right now I have no devices with the fan for live testing, so the following section
+    # generated based on device specitifations
+
+    @command(
+        default_output=format_output("Fan powering on"),
+    )
+    def fan_on(self):
+        """Power fan on (only for models with fan)."""
+        if self.model in MODELS_WITH_FAN_WY or self.model in MODELS_WITH_FAN_WY2:
+            return self.set_property("fan_power", True)
+        else:
+            raise HuizuoException("Your device doesn't support a fan management")
```

Comment:
No need for elses for exception handling, as there's a return for supported devices already in place.

This will make the code a bit cleaner to read (especially on some other cases below).

```suggestion

        raise HuizuoException("Your device doesn't support a fan management")
```

Reply:
Simplified